apollo-router/src/protocols/websocket.rs

```diff
@@ -414,11 +431,32 @@ where
                 },
                 None => Poll::Ready(None),
             },
-            Poll::Pending => Poll::Pending,
+            Poll::Pending => {
+                if let Some(heartbeat_interval) = this.heartbeat_interval {
+                    match heartbeat_interval.poll_tick(cx) {
+                        Poll::Ready(_) => send_heartbeat(this.stream, cx),
+                        Poll::Pending => (),
+                    };
+                }
+                Poll::Pending
+            }
         }
     }
 }
 
+fn send_heartbeat(mut stream: Pin<&mut impl Sink<ClientMessage>>, cx: &mut std::task::Context<'_>) {
+    if stream.as_mut().poll_flush(cx).map(|result| result.is_ok()) == Poll::Ready(true)
+        && stream.as_mut().poll_ready(cx).map(|result| result.is_ok()) == Poll::Ready(true)
+    {
+        // Ignore error
+        let _ = stream.start_send(ClientMessage::Ping {
+            payload: Some(serde_json_bytes::Value::String(
+                "APOLLO_ROUTER_HEARTBEAT".into(),
```

Comment:
Should we provide a payload ? If we can avoid sending more data than necessary I would vote for it

Reply:
It makes debugging easier, without it. 
It's not clear that this is a heartbeat.
Then I looked in traffic I also see RFC6455 PING/PONG so payload makes this particular message to standout.
I think it would also be helpful for customer's engeeneer trying to look into WS traffic.
With default interval of 5s it doesn't add that much overhead.

But this is not a principal issue for me so can remove it if you still have concerns?

Reply:
No I think you're right, if it improves the experience let's keep it